flaskr/tweets.py

```
@@ -56,12 +56,46 @@ def init_index():
     """
     tweets = get_db().session.query(Tweet).order_by(Tweet.id.desc()).all()
     for tweet in tweets:
-        content = tweet.content + " " + tweet.title
-        for word in content.split(" "):
-            if word.isalnum():  # so we can search for number too if we want
-                word = word.lower()
-                try:
-                    tweet_index[word].append(tweet.id)
-                except KeyError:
-                    tweet_index[word] = [tweet.id]
+        update_index(tweet)
     return None
+
+
+def update_index(tweet: Tweet):
+    content = tweet.content + " " + tweet.title
+    for word in content.split(" "):
+        if word.isalnum():  # so we can search for number too if we want
+            word = word.lower()
+            try:
+                tweet_index[word].append(tweet.id)
+            except KeyError:
+                tweet_index[word] = [tweet.id]
+    return None
+
+
+@bp.route("/new_tweet", methods=["GET", "POST"])
+@login_required
+def add_new_tweet():
+    if request.method == "GET":
+        return
+    # For test purposes
+    user_id = session["user_id"]
+    print(request.form.to_dict())
+    title = request.form["title"]
+    content = request.form["content"]
+    print(user_id, title, content)
+    # Untested because it crashes before
+    db = get_db()
+    author = db.session.query(User.id).filter(User.username == user_id).all()  # might miss some [0][0]
+    try:
+        new_tweet = Tweet(
+            uid=author,
+            title=title,
+            content=content,
+        )
+        db.session.add(new_tweet)
+        db.session.commit()
+    except db.IntegrityError:
+        print("DB intergrity error")
+        return redirect(url_for("tweets.index"))
```

maybe we want to return an error here, 
```return 'DB integrity error', 505```